### PR TITLE
mgr/cephadm: add support for prometheus web admin API

### PR DESCRIPTION
**Pull request by adk3798**
Allows the user to enable the web admin API through
the prometheus service spec.

Fixes: https://tracker.ceph.com/issues/63983

Signed-off-by: Adam King <adking@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/cephadm/services/monitoring.rst [hunk 1]
```diff
@@ -392,6 +392,29 @@ In the following example spec we set the retention time to 1 year and the size t
   tell cephadm to redeploy the Prometheus daemon(s) to put this change into effect.
   This can be done with a ``ceph orch redeploy prometheus`` command.
 
+Enabling Web Admin API
+~~~~~~~~~~~~~~~~~~~~~~
+
+Cephadm supports enabling the web admin api by specifying the ``enable_admin_api``
```
Comment (anthonyeleven) [inline doc/cephadm/services/monitoring.rst:398]: API

File: doc/cephadm/services/monitoring.rst [hunk 2]
```diff
@@ -392,6 +392,29 @@ In the following example spec we set the retention time to 1 year and the size t
...
+field as true in the Prometheus service spec.
```
Comment (anthonyeleven) [inline doc/cephadm/services/monitoring.rst:399]: backticks around ``true``

File: doc/cephadm/services/monitoring.rst [hunk 3]
```diff
@@ -419,6 +419,29 @@ In the following example spec we set the retention time to 1 year and the size t
...
+
+.. code-block:: yaml
+
+    service_type: prometheus
+    placement:
+      count: 1
+    spec:
+      enable_admin_api: true
+
+Like with retention time and size, this requires running ``ceph orch redeploy prometheus``
```
Comment (anthonyeleven) [inline doc/cephadm/services/monitoring.rst:436]: As with

File: doc/cephadm/services/monitoring.rst [hunk 4]
```diff
@@ -419,6 +419,29 @@ In the following example spec we set the retention time to 1 year and the size t
...
+for the setting to be picked up.
+
+.. note::
+
+  There are security concerns with enabling the admin API where anyone with
+  access to the HTTP endpoint may run arbitrary SQL against the Prometheus database.
+  Please see https://prometheus.io/docs/operating/security/#prometheus for more information
+
```
Comment (anthonyeleven) [inline doc/cephadm/services/monitoring.rst:444]: Maybe add that complex queries may cause the daemon to slow down or crash.

**Review by anthonyeleven — APPROVED** [review]

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-03-16_05:03:17-orch:cephadm-wip-adk-testing-2024-03-15-1513-distro-default-smithi/

mostly cluster log stuff + mds_upgrade_sequence. Other failures were either known or not reproducible.

**Comment by adk3798** [comment]
jenkins test api

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-03-16_05:03:17-orch:cephadm-wip-adk-testing-2024-03-15-1513-distro-default-smithi/

Tons of cluster log failures, not a problem for now. Other than that, it was just the mds_upgrade_sequence, staggered upgrade with agent (both of which are known problems), some failures installing things at the beginning of tests (known to happen occasionally), and a failure in the test_cephadm test related to idmap overrides for nfs, which I missed when we tested that PR originally `ERROR: required file missing from config-json: idmap.conf` but is not being caused by any PRs in this run.

Nothing to block merging PRs from this run.

**Pull request closed without merging**